### PR TITLE
Custom page error fix

### LINKED ISSUE #6250: Custom landing page to library breaks uwazi
**Describe the bug**
Configuring a custom landing page to a library view results in error when loading uwazi.

**To Reproduce**
1. Create a blank state uwazi
2. Go to settings -> collection
3. Select "Use custom landing page" and add one of the following landing pages (im including more than one because the errors are different, not sure if this is the same issue or a different one)
  - "/en/library/?q=(allAggregations:!t,includeUnpublished:!t,order:desc,sort:creationDate,treatAs:number,unpublished:!f)"
  - "/en/library/"
4. Go to localhost:3000
5. See errors

**Expected behavior**
Library as a landing page should work properly, or maybe have a fallback in case of errors ? so that we can navigate to settings and fix the potential typo/error.

**Additional context**
On the first case "/en/library/?q=(allAggregations:!t,includeUnpublished:!t,order:desc,sort:creationDate,treatAs:number,unpublished:!f)" you can manually go to /settings deactivate the custom landing page and fix the instance.

Only on the second case "/en/library/" the error happens no matter what url you visit (this seems strange, im assuming the landing page code should only be executed when navigating to root ?)

### PR DESCRIPTION
**Pull request by grafitto**
fixes #6250 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/react/entry-server.tsx [hunk 1]
```diff
@@ -193,6 +193,12 @@ const setReduxState = async (
       headers
     );
 
+    // @ts-ignore
```
Comment (Zasa-san) [inline app/react/entry-server.tsx]: Please check if we can use types in the few `@ts-ignore`s.

File: app/react/entry-server.tsx [hunk 2]
```diff
@@ -206,6 +212,7 @@ const setReduxState = async (
       );
     } catch (e) {
       if (e instanceof FetchResponseError) {
+        // console.log(e.json.errors);
```
Comment (Zasa-san) [inline app/react/entry-server.tsx]: Please remove the logs, there's a couple of them :grimacing:

File: e2e/suite1/pages.test.ts [hunk 3]
```diff
@@ -132,4 +132,37 @@ describe('Custom home page and styles', () => {
     const rowSelector = '.tableview-wrapper > table > tbody > tr.template-58ada34c299e82674854504b';
     expect((await page.$$(rowSelector)).length).toBe(2);
   });
+
+  it('should allow using a complicated library query as a landing page', async () => {
```
Comment (Zasa-san) [inline e2e/suite1/pages.test.ts:136]: Is there a unit test where you could test with more diverse cases? It's OK to have it covered by an E2E, but in my opinion this type of things should be tested in the E2E in a rather simple way, and heavily tested in unit tests. That said, I defer to your opinion on this.

File: app/react/getIndexElement.tsx [hunk 4]
```diff
@@ -11,7 +11,16 @@ import { ViewerRoute } from './Viewer/ViewerRoute';
 
 const getCustomLibraryPage = (customHomePage: string[]) => {
   const [query] = customHomePage.filter(path => path.startsWith('?'));
-  const queryString = `(${query.substring(1)})`;
+  let searchQuery = '';
+  if (query) {
```
Comment (Zasa-san) [inline app/react/getIndexElement.tsx]: Codeclimate is complaining about this, maybe we can refactor the nested `if`s a bit

**Review by Zasa-san — CHANGES_REQUESTED** [review]
The bug seems fixed! Nice!

File: app/react/entry-server.tsx (hunk 1 above)
Comment (grafitto) [reply app/react/entry-server.tsx]: Removed

File: app/react/entry-server.tsx (hunk 2 above)
Comment (grafitto) [reply app/react/entry-server.tsx]: Removed

File: app/react/getIndexElement.tsx (hunk 4 above)
Comment (grafitto) [reply app/react/getIndexElement.tsx]: Fixed

Comment (grafitto) [reply app/react/getIndexElement.tsx]: Done

**Comment by grafitto** [comment]
Just realized this fixes a critical issue. Will create another how fix then close this.

**Comment by grafitto** [comment]
> Just realized this fixes a critical issue. Will create another how fix then close this.

https://github.com/huridocs/uwazi/pull/6402 here is the hotfix

**Pull request closed without merging**